app/src/org/commcare/network/HttpCalloutTask.java

```diff
@@ -116,6 +125,16 @@ protected HttpCalloutOutcomes doTaskBackground(Object... params) {
         return doPostCalloutTask(calloutFailed);
     }
 
+    protected boolean isCaptivePortal() throws IOException {
```

Comment:
This definitely is a better check for captive portal than what we do in `ConnectionDiagnosticTask`. We should abstract this function to  `ConnectivityStatus` and use it at both places.